src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -48219,6 +48219,243 @@ static void F(bool x, bool y, bool z, bool? w)
+                //         object a = (Func<string>)(b ? () => s.ToString() : () => s?.ToString());
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "s?.ToString()").WithLocation(8, 66)
+            );
+        }
+
+        [Fact, WorkItem(51461, "https://github.com/dotnet/roslyn/issues/51461")]
+        public void ConditionalOperator_TargetTyped_03()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class C
+{
+    static void M(bool b)
+    {
+        Func<string>? s = () => """";
+        Func<object>? o = b ? () => s() : s = null;
+    }
+}
+", options: WithNullableEnable());
+
+            comp.VerifyDiagnostics(
+            );
+        }
+
+        [Fact, WorkItem(51461, "https://github.com/dotnet/roslyn/issues/51461")]
+        public void ConditionalOperator_TargetTyped_04()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class C
+{
+    static void M(bool b)
+    {
+        Func<string>? s = () => """";
+        Func<object>? o = b ? s = null : () => s();
+    }
+}
+", options: WithNullableEnable());
+
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(51461, "https://github.com/dotnet/roslyn/issues/51461")]
+        public void ConditionalOperator_TargetTyped_05()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    static void M(bool b)
+    {
+        string? s = null;
+        object a = b ? () => s.ToString() : () => s?.ToString();
+    }
+}
+", options: WithNullableEnable());
+
+            comp.VerifyDiagnostics(
+                // (7,24): error CS1660: Cannot convert lambda expression to type 'object' because it is not a delegate type
+                //         object a = b ? () => s.ToString() : () => s?.ToString();
+                Diagnostic(ErrorCode.ERR_AnonMethToNonDel, "() => s.ToString()").WithArguments("lambda expression", "object").WithLocation(7, 24),
+                // (7,30): warning CS8602: Dereference of a possibly null reference.
+                //         object a = b ? () => s.ToString() : () => s?.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(7, 30),
+                // (7,45): error CS1660: Cannot convert lambda expression to type 'object' because it is not a delegate type
+                //         object a = b ? () => s.ToString() : () => s?.ToString();
+                Diagnostic(ErrorCode.ERR_AnonMethToNonDel, "() => s?.ToString()").WithArguments("lambda expression", "object").WithLocation(7, 45)
+            );
+        }
+
+        [Fact, WorkItem(51461, "https://github.com/dotnet/roslyn/issues/51461")]
+        public void ConditionalOperator_TargetTyped_06()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class C
+{
+    static void M(bool b)
+    {
+        string? s1 = null;
+        string? s2 = """";
+        M1(s2, b ? () => s1.ToString() : () => s1?.ToString()).ToString();
```

Comment:
Is it just impossible to infer `string?` here for `T` in combination with target typed switch/conditional? It feels like if the argument were `() => s1?.ToString()`, we would infer `string?` here.